app/models/pet.rb

```diff
@@ -95,6 +97,12 @@ def append_images=(attachables)
     images.attach(attachables)
   end
 
+  def sensible_placement_type
+    if matches.where(end_date: DateTime.now..).exists? && placement_type == "Adoptable"
+      errors.add(:placement_type, "must include Fosterable")
```

Comment:
Yes I think we can make this error more explicit. Something like 'Cannot change placement type to adoptable on pet with active foster'

Also - please add this to the en.yml and use translation. I think you need to call the full i18n method to access in the model.